publisher-subscriber/README.md

```diff
@@ -0,0 +1,358 @@
+            //Adds event listener to subscriber and uses onMessage as a callback
+            subscriber.setMessageListener(this);
+
+            tConnection.start();
+            System.out.println("Initial rate is " + currentRate + " \nWaiting for new rates...");
+        } catch(NamingException e) {
+            e.printStackTrace();
+            LOGGER.error("An error has occurred!", e);
+            System.exit(1);
+        } catch(JMSException e) {
+            e.printStackTrace();
+            LOGGER.error("An error has occurred!", e);
+            System.exit(1);
+        }
+    }
+
+    //This method is called asynchronously by the activeMQ broker
+    public void onMessage(Message message) {
+
+        try {
+            BytesMessage bMessage = (BytesMessage) message;
+            double newRate = ((BytesMessage) message).readDouble();
+
+            if (currentRate - newRate >= 1)
+                System.out.println("New Rate is " + newRate + " - Consider refinancing");
+            else
+                System.out.println("New Rate is " + newRate + " - Consider keeping current rate");
+        } catch(JMSException e) {
+            e.printStackTrace();
+            LOGGER.error("An error occurred!", e);
+            System.exit(1);
+        }
+        System.out.println("Waiting for new rates...");
+    }
+
+    private void exit() {
+        try {
+            tConnection.close();
+        } catch(JMSException e) {
+            e.printStackTrace();
+            LOGGER.error("An error has occurred!", e);
+            System.exit(1);
+        }
+        System.exit(0);
+    }
+
+    public static void main(String[] args) {
+
+        String topicCF = null;
+        String topicName = null;
+        int rate = 0;
+        if (args.length == 3) {
+            topicCF = args[0];
+            topicName = args[1];
+            rate = Integer.parseInt(args[2]);
+        } else {
+            System.out.println("Invalid arguments. Should be: ");
+            System.out.println("java TBorrower [factory] [topic] [rate]");
+            System.exit(0);
+        }
+
+        TBorrower tBorrower = new TBorrower(topicCF, topicName, rate);
+
+        try {
+            // Run until enter is pressed
+            BufferedReader reader = new BufferedReader
+                    (new InputStreamReader(System.in));
+            System.out.println ("TBorrower application started");
+            System.out.println ("Press enter to quit application");
+            reader.readLine();
+            tBorrower.exit();
+        } catch (IOException ioe) {
+            ioe.printStackTrace();
+        }
+    }
+
+    public TopicConnection gettConnection() {
+        return tConnection;
+    }
+
+    public TopicSession gettSession() {
+        return tSession;
+    }
+
+    public Topic getTopic() {
+        return topic;
+    }
+
+    public double getCurrentRate() {
+        return currentRate;
+    }
+}
+```
+
+## Class diagram
```

Comment:
Class diagram is missing. Use urm-maven-plugin for that...

Reply:
I tried to get this to work, I can't seem to figure it out...perhaps this newest commit will show what I am doing wrong.

Reply:
The first step is to include your module in the parent pom.xml. After that, urm-maven-plugin should automatically process the module when doing maven build.